userspace/libsinsp/container.cpp

```diff
@@ -1293,6 +1359,67 @@ string sinsp_container_manager::get_container_name(sinsp_threadinfo* tinfo)
 	return res;
 }
 
+void sinsp_container_manager::identify_healthcheck(sinsp_threadinfo *tinfo)
+{
+	// This thread is a part of a container healthcheck if its
+	// parent thread is part of a health check.
+	sinsp_threadinfo* ptinfo = tinfo->get_parent_thread();
+
+	if(ptinfo && ptinfo->m_is_container_healthcheck)
+	{
+		tinfo->m_is_container_healthcheck = true;
+		return;
+	}
+
+	sinsp_container_info *cinfo = get_container(tinfo->m_container_id);
+
+	if(!cinfo)
+	{
+		return;
+	}
+
+	// Otherwise, the thread is a part of a container healthcheck if:
+	//
+	// 1. the comm and args match the container's healthcheck
```

Comment:
Why isn't technique 2 enough? My only concern with 1 is that there's a (probably remote, I agree) chance of FPs.